abel/tools/transform_pairs.py

```diff
@@ -0,0 +1,146 @@
+from __future__ import division
+import numpy as np
+
+##############################################################################
+#
+# Analytical Abel transform pairs: profiles 1-7
+#  G. C.-Y Chan and G. M. Hieftje Spectrochimica Acta B 61, 31-41 (2006)
+#  doi:10.1016/j.sab.2005.11.009
+#
+# 20-Dec-2017 Stephen Gibson - adapted code for PyAbel
+# 20-Nov-2015 Dhrubajyoti Das - python gist
+#             https://github.com/PyAbel/PyAbel/issues/19#issuecomment-158244527
+#
+# Note: call these functions via the class method:
+#   func = abel.tools.analytical.TransformPair(n, profile=#)
+#   see abel/tools/analytical.py for Class attributes
+# 
+##############################################################################
+
+def a(n, x):
+    return np.sqrt(n*n - x*x)
+
+
+def profile1(r):
+    # Cremers and Birkebak App. Opt. 5, 1057-1064 (1966) Eq(13)
```

Comment:
Can we change these comments to docstrings, so that they will be included in the documentation? I think it would also be nice to have a one sentence summary of that the function looks like. Something as simple as "Starts at 1 at r=0 and decays slowly to 0 at r -> infinity."

I think that you are assuming that people won't want to use these directly, since it's probably better to use them through abel.tools.analytical.TransformPair, but it seems possible that people may want to use these directly. 

You may want to add the line about "call this function using abel.tools.analytical.TransformPair" to each docstring. I suppose I'm advocating to append an identical few-line docstring to each of thees functions, with a unique sentence for each.

Reply:
Good comments. 
1. Transform pairs may now be accessed directly via `abel.tools.transform_pairs.profile#(r)` where `# = 1-8`.
2. Profile functions will now accept a single value for r.
3. Added r range check to ensure  `0 <= r <=1`, although a few of the functions require 0 < r <= 1 or 0 < r < 1. This is checked, but the docs may need improvement to address this subtly. 
Items 1-3 are not required for the `abel.tools.analytical.TransformPair` class, since it generates `r=np.linspace(0, 1, n)` and slightly, offsets the limits to prevent NaNs.
4. Added docstrings to each profile and ascii plots(*) of the source and projection functions, as a picture is worth a 1000 words (and these profiles are not easy to describe).

(*) ascii plots from the [AP package](https://github.com/mfouesneau/asciiplot)